### PR TITLE
Deprecate `plone.app.widgets`

### PR DESCRIPTION
**Pull request by petschki**
~~this needs https://github.com/plone/plone.app.z3cform/pull/155~~ merged 👍🏼

### CODE REVIEW

**Comment by petschki** [comment]
After sleeping over this, I do not think it's the correct way to import from `plone.app.z3cform` ... maybe we should move the pattern configuration code out of CMFPlone to `plone.base`? Or the other way round: implement pattern specific configuration here and let other packages reuse it ?

**Review by jensens — APPROVED** [review]

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Review by jensens — COMMENTED** [review]
actually, the base idea is good and worth a try in my opinion, but maybe more eyes on this can help to get it done right?

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

File: setup.py [hunk 1]
```diff
@@ -75,6 +75,7 @@
         "plone.app.viewletmanager",
         "plone.app.vocabularies",
         "plone.app.workflow",
+        "plone.app.z3cform >= 4.1.0",
```
Comment (petschki) [inline setup.py:78]: @jensens is it ok to define dependency here? Maybe this causes more circular import troubles ...

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

File: setup.py (hunk 1 above)
Comment (jensens) [reply setup.py:78]: This is probably a problem. Need to check. A good little task for ArtSprint!

Comment (mauritsvanrees) [reply setup.py:78]: This PR was merged, but was this conversation resolved somewhere?

Comment (petschki) [reply setup.py:78]: Sorry for silence ... @jensens looked at it here at ArtSprint and it didn't have any circular issues.

**Pull request merged**